src/message.rs

```diff
@@ -517,6 +517,12 @@ impl OwnedMessage {
             headers,
         }
     }
+
+    /// Detach `OwnedHeaders` from OwnedMessage.
+    /// If you just need a reference, peek the headers() method.
```

Comment:
I think we can omit this second sentence, and link to the `OwnedHeaders`:
```suggestion
    /// Detaches the [`OwnedHeaders`] from this `OwnedMessage`.
```